src/internal/components/LayoutEditor.tsx

```diff
@@ -0,0 +1,234 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { LoadingScreen } from "../puck/components/LoadingScreen.tsx";
+import { InternalLayoutEditor } from "./InternalLayoutEditor.tsx";
+import { InitialHistory, Config } from "@measured/puck";
+import { LayoutSaveState } from "../types/saveState.ts";
+import { TemplateMetadata } from "../types/templateMetadata.ts";
+import { useMessageSenders } from "../hooks/useMessage.ts";
+import { useLocalStorage } from "../hooks/useLocalStorage.ts";
+import { DevLogger } from "../../utils/devLogger.ts";
+import { jsonFromEscapedJsonString } from "../utils/jsonFromEscapedJsonString.ts";
+
+const devLogger = new DevLogger();
+
+type LayoutEditorProps = {
+  puckConfig: Config;
+  templateMetadata: TemplateMetadata;
+  layoutSaveState: LayoutSaveState | undefined;
+  visualConfigurationData: any;
+};
+
+export const LayoutEditor = (props: LayoutEditorProps) => {
+  const {
+    puckConfig,
+    templateMetadata,
+    layoutSaveState,
+    visualConfigurationData,
+  } = props;
+
+  const {
+    sendDevLayoutSaveStateData,
+    saveLayoutSaveState,
+    publishVisualConfiguration,
+    deleteLayoutSaveState,
+  } = useMessageSenders();
```

Comment:
What about breaking this out even more?

/hooks/messageSenders -> export commonMessageSenders()
/hooks/layout/messageSenders.ts -> export layoutMessageSenders()
/hooks/theme/messageSenders.ts -> export themeMessageSenders()

I could see naming the files like common/layout/themeMessageSenders.ts but then it's redundant, but helps with opening files when 3 have the same name.

But this would at least prevent us from creating senders (and receivers if you do the same) for things not necessary in layout vs theme.